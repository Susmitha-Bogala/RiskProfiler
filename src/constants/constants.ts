export const questions = [
    {
      id: 1,
      text: 'How would you describe your investment knowledge?',
      options: [
        { label: 'Novice', score: 1 },
        { label: 'Intermediate', score: 2 },
        { label: 'Advanced', score: 3 },
      ],
    },
    {
      id: 2,
      text: 'Investment duration?',
      options: [
        { label: 'Short-term (less than 1 year)', score: 1 },
        { label: 'Medium-term (1-5 years)', score: 2 },
        { label: 'Long-term (more than 5 years)', score: 3 },
      ],
    },
    {
      id: 3,
      text: 'How comfortable are you taking risks?',
      options: [
        { label: 'Very risk averse', score: 1 },
        { label: 'Somewhat risk averse', score: 2 },
        { label: 'Neutral', score: 3 },
        { label: 'Somewhat risk-tolerant', score: 4 },
        { label: 'Very risk-tolerant', score: 5 },
      ],
    },
    {
      id: 4,
      text: 'What percentage of your income are you willing to invest?',
      options: [
        { label: 'Less than 10%', score: 1 },
        { label: '10-25%', score: 2 },
        { label: '25-50%', score: 3 },
        { label: 'More than 50%', score: 5 },
      ],
    },
    {
      id: 5,
      text: 'How would you react to a sudden drop in investments?',
      options: [
        { label: 'Panic and sell immediately', score: 1 },
        { label: 'Monitor closely and consider selling', score: 2 },
        { label: 'Hold and wait for recovery', score: 3 },
        { label: 'See it as buying opportunity', score: 5 },
      ],
    },
  ];

export const riskDescriptions = {
  Low: 'You prefer safer investments and value capital protection.',
  Medium: 'You are moderately open to risk for better returns.',
  High: 'You are comfortable taking higher risks for higher potential gains.',
};

export const riskType = 'Risk Type:';

export const riskScore = 'Total Score:';
  