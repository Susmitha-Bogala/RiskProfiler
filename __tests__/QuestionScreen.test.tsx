import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import QuestionScreen from '../src/screens/QuestionScreen';
import { questions } from '../src/constants/constants';
import { NavigationContainer } from '@react-navigation/native';
import { useRoute } from '@react-navigation/native';
import { useNavigation } from '@react-navigation/native';
import styles from '../src/styles/globalStyles';

jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useRoute: jest.fn(),
}));

const mockStore = configureStore([]);

const renderWithProviders = (ui, { reduxState } = {}) => {
  const store = mockStore(reduxState);
  return render(
    <Provider store={store}>
      <NavigationContainer>{ui}</NavigationContainer>
    </Provider>,
  );
};

// Define mockNavigate and mockNavigation
const mockNavigate = jest.fn();
const mockNavigation = {
  navigate: mockNavigate,
};

// Mock useNavigation hook
jest
  .spyOn(require('@react-navigation/native'), 'useNavigation')
  .mockReturnValue(mockNavigation);

describe('QuestionScreen', () => {
  beforeEach(() => {
    // Mock the useRoute hook to return specific parameters
    useRoute.mockReturnValue({
      params: { questionId: 1 },
    });
  });

  it('renders question and options correctly and handles option selection', () => {
    const initialState = { risk: { answers: [] } };
    const store = mockStore(initialState);
    const { getByText, getByTestId, toJSON } = render(
      <Provider store={store}>
        <NavigationContainer>
          <QuestionScreen />
        </NavigationContainer>
      </Provider>,
    );

    // Verify the question text is rendered
    expect(getByText(`Question 1 of ${questions.length}`)).toBeTruthy();
    expect(getByText(questions[0].text)).toBeTruthy();

    // Verify each option is rendered and can be interacted with
    questions[0].options.forEach((option, idx) => {
      const optionButton = getByTestId(`option-${idx}`);
      expect(optionButton).toBeTruthy();

      // Simulate pressing the option
      fireEvent.press(optionButton);

      // Verify that the option selection triggers the correct action
      const actions = store.getActions();
      expect(actions).toContainEqual({
        type: 'risk/setAnswer',
        payload: { questionId: 1, score: option.score },
      });
    });

    // Snapshot test
    expect(toJSON()).toMatchSnapshot();
  });

  it('navigates to the next question if not the last question', async () => {
    const initialState = { risk: { answers: [] } };
    const { getByText, toJSON } = renderWithProviders(<QuestionScreen />, {
      reduxState: initialState,
    });

    const firstOption = questions[0].options[0];

    fireEvent.press(getByText(firstOption.label));

    // Update the mock to simulate navigation to the next question
    useRoute.mockReturnValue({
      params: { questionId: 2 },
    });

    // Re-render the component with the updated route parameters
    const { getByText: getByText2 } = renderWithProviders(<QuestionScreen />, {
      reduxState: initialState,
    });

    await waitFor(() => {
      expect(getByText2(`Question 2 of ${questions.length}`)).toBeTruthy();
    });

    // Snapshot test
    expect(toJSON()).toMatchSnapshot();
  });

  it('navigates to the Result screen if it is the last question', async () => {
    const initialState = { risk: { answers: [] } };

    const lastQuestionId = questions[questions.length - 1].id;

    const lastOption = questions[lastQuestionId - 1].options[0];

    // Update the mock to simulate the last question
    useRoute.mockReturnValue({
      params: { questionId: lastQuestionId },
    });

    const { getByText, toJSON } = renderWithProviders(<QuestionScreen />, {
      reduxState: initialState,
    });

    fireEvent.press(getByText(lastOption.label));

    // Mock navigation to Result
    const mockNavigate = useNavigation().navigate;

    await waitFor(() => {
      expect(mockNavigate).toHaveBeenCalledWith('Result');
    });

    // Snapshot test
    expect(toJSON()).toMatchSnapshot();
  });

  it('updates the Redux state on answer selection', () => {
    const initialState = { risk: { answers: [] } };
    const store = mockStore(initialState);
    const { getByText, toJSON } = render(
      <Provider store={store}>
        <NavigationContainer>
          <QuestionScreen />
        </NavigationContainer>
      </Provider>,
    );
    const firstOption = questions[0].options[0];
    fireEvent.press(getByText(firstOption.label));

    const actions = store.getActions();
    expect(actions).toContainEqual({
      type: 'risk/setAnswer',
      payload: { questionId: 1, score: firstOption.score },
    });

    // Snapshot test
    expect(toJSON()).toMatchSnapshot();
  });

  it('applies correct styling to selected option', () => {
    const initialState = {
      risk: {
        answers: [{ questionId: 1, score: questions[0].options[0].score }],
      },
    };
    const { getByTestId, toJSON } = renderWithProviders(<QuestionScreen />, {
      reduxState: initialState,
    });

    const selectedOption = getByTestId('option-0');
    expect(selectedOption.props.style).toMatchObject(styles.optionSelected);

    // Snapshot test
    expect(toJSON()).toMatchSnapshot();
  });

  it('returns null when no question is found for the given questionId', () => {
    const initialState = { risk: { answers: [] } };

    // Mock the useRoute hook to return a non-existent questionId
    useRoute.mockReturnValue({
      params: { questionId: -1 },
    });

    const { toJSON } = renderWithProviders(<QuestionScreen />, {
      reduxState: initialState,
    });

    // Verify that the component returns null
    expect(toJSON()).toBeNull();

    expect(toJSON()).toMatchSnapshot();
  });
});
