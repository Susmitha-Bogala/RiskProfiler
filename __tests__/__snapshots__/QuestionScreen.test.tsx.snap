// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionScreen applies correct styling to selected option 1`] = `
<View
  collapsable={false}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#f5fbff",
      "flex": 1,
      "justifyContent": "center",
      "opacity": 0,
      "paddingHorizontal": 24,
    }
  }
>
  <Text
    style={
      {
        "alignSelf": "flex-start",
        "color": "#555",
        "fontSize": 16,
        "marginBottom": 12,
      }
    }
  >
    Question 
    1
     of 
    5
  </Text>
  <View
    style={
      {
        "backgroundColor": "#ffffff",
        "borderRadius": 16,
        "elevation": 6,
        "padding": 24,
        "shadowColor": "#000",
        "shadowOpacity": 0.1,
        "shadowRadius": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#333",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 20,
        }
      }
    >
      How would you describe your investment knowledge?
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#e6f0ff",
          "borderColor": "#007BFF",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 12,
          "opacity": 1,
          "padding": 14,
        }
      }
      testID="option-0"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#999",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "marginRight": 14,
              "width": 20,
            },
            {
              "backgroundColor": "#007BFF",
              "borderColor": "#007BFF",
            },
          ]
        }
      />
      <Text
        style={
          {
            "color": "#333",
            "fontSize": 16,
          }
        }
      >
        Novice
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "#ccc",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 12,
          "opacity": 1,
          "padding": 14,
        }
      }
      testID="option-1"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#999",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "marginRight": 14,
              "width": 20,
            },
            false,
          ]
        }
      />
      <Text
        style={
          {
            "color": "#333",
            "fontSize": 16,
          }
        }
      >
        Intermediate
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "#ccc",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 12,
          "opacity": 1,
          "padding": 14,
        }
      }
      testID="option-2"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#999",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "marginRight": 14,
              "width": 20,
            },
            false,
          ]
        }
      />
      <Text
        style={
          {
            "color": "#333",
            "fontSize": 16,
          }
        }
      >
        Advanced
      </Text>
    </View>
  </View>
</View>
`;

exports[`QuestionScreen navigates to the Result screen if it is the last question 1`] = `
<View
  collapsable={false}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#f5fbff",
      "flex": 1,
      "justifyContent": "center",
      "opacity": 0,
      "paddingHorizontal": 24,
    }
  }
>
  <Text
    style={
      {
        "alignSelf": "flex-start",
        "color": "#555",
        "fontSize": 16,
        "marginBottom": 12,
      }
    }
  >
    Question 
    5
     of 
    5
  </Text>
  <View
    style={
      {
        "backgroundColor": "#ffffff",
        "borderRadius": 16,
        "elevation": 6,
        "padding": 24,
        "shadowColor": "#000",
        "shadowOpacity": 0.1,
        "shadowRadius": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#333",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 20,
        }
      }
    >
      How would you react to a sudden drop in investments?
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "#ccc",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 12,
          "opacity": 1,
          "padding": 14,
        }
      }
      testID="option-0"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#999",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "marginRight": 14,
              "width": 20,
            },
            false,
          ]
        }
      />
      <Text
        style={
          {
            "color": "#333",
            "fontSize": 16,
          }
        }
      >
        Panic and sell immediately
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "#ccc",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 12,
          "opacity": 1,
          "padding": 14,
        }
      }
      testID="option-1"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#999",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "marginRight": 14,
              "width": 20,
            },
            false,
          ]
        }
      />
      <Text
        style={
          {
            "color": "#333",
            "fontSize": 16,
          }
        }
      >
        Monitor closely and consider selling
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "#ccc",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 12,
          "opacity": 1,
          "padding": 14,
        }
      }
      testID="option-2"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#999",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "marginRight": 14,
              "width": 20,
            },
            false,
          ]
        }
      />
      <Text
        style={
          {
            "color": "#333",
            "fontSize": 16,
          }
        }
      >
        Hold and wait for recovery
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "#ccc",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 12,
          "opacity": 1,
          "padding": 14,
        }
      }
      testID="option-3"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#999",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "marginRight": 14,
              "width": 20,
            },
            false,
          ]
        }
      />
      <Text
        style={
          {
            "color": "#333",
            "fontSize": 16,
          }
        }
      >
        See it as buying opportunity
      </Text>
    </View>
  </View>
</View>
`;

exports[`QuestionScreen navigates to the next question if not the last question 1`] = `
<View
  collapsable={false}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#f5fbff",
      "flex": 1,
      "justifyContent": "center",
      "opacity": 0,
      "paddingHorizontal": 24,
    }
  }
>
  <Text
    style={
      {
        "alignSelf": "flex-start",
        "color": "#555",
        "fontSize": 16,
        "marginBottom": 12,
      }
    }
  >
    Question 
    1
     of 
    5
  </Text>
  <View
    style={
      {
        "backgroundColor": "#ffffff",
        "borderRadius": 16,
        "elevation": 6,
        "padding": 24,
        "shadowColor": "#000",
        "shadowOpacity": 0.1,
        "shadowRadius": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#333",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 20,
        }
      }
    >
      How would you describe your investment knowledge?
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "#ccc",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 12,
          "opacity": 1,
          "padding": 14,
        }
      }
      testID="option-0"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#999",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "marginRight": 14,
              "width": 20,
            },
            false,
          ]
        }
      />
      <Text
        style={
          {
            "color": "#333",
            "fontSize": 16,
          }
        }
      >
        Novice
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "#ccc",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 12,
          "opacity": 1,
          "padding": 14,
        }
      }
      testID="option-1"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#999",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "marginRight": 14,
              "width": 20,
            },
            false,
          ]
        }
      />
      <Text
        style={
          {
            "color": "#333",
            "fontSize": 16,
          }
        }
      >
        Intermediate
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "#ccc",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 12,
          "opacity": 1,
          "padding": 14,
        }
      }
      testID="option-2"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#999",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "marginRight": 14,
              "width": 20,
            },
            false,
          ]
        }
      />
      <Text
        style={
          {
            "color": "#333",
            "fontSize": 16,
          }
        }
      >
        Advanced
      </Text>
    </View>
  </View>
</View>
`;

exports[`QuestionScreen renders question and options correctly and handles option selection 1`] = `
<View
  collapsable={false}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#f5fbff",
      "flex": 1,
      "justifyContent": "center",
      "opacity": 0,
      "paddingHorizontal": 24,
    }
  }
>
  <Text
    style={
      {
        "alignSelf": "flex-start",
        "color": "#555",
        "fontSize": 16,
        "marginBottom": 12,
      }
    }
  >
    Question 
    1
     of 
    5
  </Text>
  <View
    style={
      {
        "backgroundColor": "#ffffff",
        "borderRadius": 16,
        "elevation": 6,
        "padding": 24,
        "shadowColor": "#000",
        "shadowOpacity": 0.1,
        "shadowRadius": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#333",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 20,
        }
      }
    >
      How would you describe your investment knowledge?
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "#ccc",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 12,
          "opacity": 1,
          "padding": 14,
        }
      }
      testID="option-0"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#999",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "marginRight": 14,
              "width": 20,
            },
            false,
          ]
        }
      />
      <Text
        style={
          {
            "color": "#333",
            "fontSize": 16,
          }
        }
      >
        Novice
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "#ccc",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 12,
          "opacity": 1,
          "padding": 14,
        }
      }
      testID="option-1"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#999",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "marginRight": 14,
              "width": 20,
            },
            false,
          ]
        }
      />
      <Text
        style={
          {
            "color": "#333",
            "fontSize": 16,
          }
        }
      >
        Intermediate
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "#ccc",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 12,
          "opacity": 1,
          "padding": 14,
        }
      }
      testID="option-2"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#999",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "marginRight": 14,
              "width": 20,
            },
            false,
          ]
        }
      />
      <Text
        style={
          {
            "color": "#333",
            "fontSize": 16,
          }
        }
      >
        Advanced
      </Text>
    </View>
  </View>
</View>
`;

exports[`QuestionScreen returns null when no question is found for the given questionId 1`] = `null`;

exports[`QuestionScreen updates the Redux state on answer selection 1`] = `
<View
  collapsable={false}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#f5fbff",
      "flex": 1,
      "justifyContent": "center",
      "opacity": 0,
      "paddingHorizontal": 24,
    }
  }
>
  <Text
    style={
      {
        "alignSelf": "flex-start",
        "color": "#555",
        "fontSize": 16,
        "marginBottom": 12,
      }
    }
  >
    Question 
    1
     of 
    5
  </Text>
  <View
    style={
      {
        "backgroundColor": "#ffffff",
        "borderRadius": 16,
        "elevation": 6,
        "padding": 24,
        "shadowColor": "#000",
        "shadowOpacity": 0.1,
        "shadowRadius": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#333",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 20,
        }
      }
    >
      How would you describe your investment knowledge?
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "#ccc",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 12,
          "opacity": 1,
          "padding": 14,
        }
      }
      testID="option-0"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#999",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "marginRight": 14,
              "width": 20,
            },
            false,
          ]
        }
      />
      <Text
        style={
          {
            "color": "#333",
            "fontSize": 16,
          }
        }
      >
        Novice
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "#ccc",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 12,
          "opacity": 1,
          "padding": 14,
        }
      }
      testID="option-1"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#999",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "marginRight": 14,
              "width": 20,
            },
            false,
          ]
        }
      />
      <Text
        style={
          {
            "color": "#333",
            "fontSize": 16,
          }
        }
      >
        Intermediate
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "borderColor": "#ccc",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 12,
          "opacity": 1,
          "padding": 14,
        }
      }
      testID="option-2"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderColor": "#999",
              "borderRadius": 10,
              "borderWidth": 2,
              "height": 20,
              "justifyContent": "center",
              "marginRight": 14,
              "width": 20,
            },
            false,
          ]
        }
      />
      <Text
        style={
          {
            "color": "#333",
            "fontSize": 16,
          }
        }
      >
        Advanced
      </Text>
    </View>
  </View>
</View>
`;
